### PR TITLE
Get python tests to pass and java tests to run 

### PR DESCRIPTION
**Pull request by MrCreosote**
Lots of failures in the java tests as expected

### CODE REVIEW

File: .gitmodules
Comment (briehl) [inline .gitmodules:1]: 🎉

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+requests = "*"
+
+[dev-packages]
+pytest = "*"
+coverage = "*"
+pytest-cov = "*"
+ipython = "*"
+
+[requires]
+python_version = "3.8"
```
Comment (briehl) [inline Pipfile:16]: Can we force it on something newer? Or not yet?

**Review by briehl — APPROVED** [review]
Looks good. No need to bump the Python version yet.

File: Pipfile (hunk 1 above)
Comment (MrCreosote) [reply Pipfile:16]: I don't think we have a base image newer than 3.8. 

I want to entirely rethink how we handle the base image but that's a way down the road.

**Pull request merged**